rust/gui-client/docs/launch-modes.md

```diff
@@ -0,0 +1,50 @@
+# Linux launch modes
+
+(Without thinking of systemd yet)
+
+1. No GUI allowed
+2. GUI spawns its own tunnel subprocess
```

Comment:
I don't understand. If we're spawning another process then we are using two processes, right? Running the GUI as root is breaking a lot of stuff, so if the GUI spawns a tunnel subprocess, that means the GUI sticks around as the non-privileged parent process.

Sudo will require a password on most systems. There is a graphical sudo for desktop environments that will kick in, sometimes I think it happens automagically even if we just call `sudo` in the app.

Yeah we should see what Docker does. "Privileged daemon / userspace client" would be Mode 3 in this doc. 

Due to DNS I think we are required to run the tunnel as root. I didn't find any more fine-grained security for setting DNS on an interface. (Other than hooking into NetworkManager) If we run as root, there's no need to do setcap.

I think we're really still talking past each other on 2 processes vs 2 binaries. We can run a privileged process and an unprivileged process from the same binary. Even if unprivileged code has a vulnerability and jumps into privileged code, it couldn't do anything since it won't have root powers.

So there is no need to package 2 binaries using Tauri bundling or anything - It's already easy to do Busybox-style multi-call binaries where two "binaries" are packed into one binary at the source code level, and I'm already doing that for the crash handling.

Like I would not be surprised if it turns out that Docker's privileged daemon and non-privileged userspace client both run from the same binary on disk. That's how web browsers do it.